src/hotspot/os/windows/permitForbiddenFunctions_windows.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef OS_WINDOWS_PERMITFORBIDDENFUNCTIONS_WINDOWS_HPP
+#define OS_WINDOWS_PERMITFORBIDDENFUNCTIONS_WINDOWS_HPP
+
+#include "utilities/compilerWarnings.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+// Provide wrappers for some functions otherwise forbidden from use in HotSpot.
+// See forbiddenFunctions.hpp for details.
+
+namespace permit_forbidden_functions {
+BEGIN_ALLOW_FORBIDDEN_FUNCTIONS
+
+// Used by the Windows implementation of os::realpath.
+inline char* _fullpath(char* absPath, const char* relPath, size_t maxLength) {
+  return ::_fullpath(absPath, relPath, maxLength);
+}
+
```

Comment:
Could we forbid the non Standard \_snprintf too?

Reply:
Sure, I'll add that.  We should also technically be forbidding snprintf in favor of os::snprintf, but the difference
is pretty minuscule and there are a significant number of occurrences (125 today).